modules/page-memory/src/main/java/org/apache/ignite/internal/pagememory/persistence/checkpoint/Checkpointer.java

```diff
@@ -194,32 +194,18 @@ protected void body() {
                 waitCheckpointEvent();
 
                 if (skipCheckpointOnNodeStop && (isCancelled() || shutdownNow)) {
-                    if (log.isInfoEnabled()) {
+                    if (log.isWarnEnabled()) {
                         log.warn("Skipping last checkpoint because node is stopping.");
                     }
 
                     return;
                 }
 
-                CompletableFuture<?> enableChangeAppliedFuture = this.enableChangeAppliedFuture;
-
-                if (enableChangeAppliedFuture != null) {
-                    enableChangeAppliedFuture.complete(null);
-
-                    this.enableChangeAppliedFuture = null;
-                }
-
-                if (checkpointsEnabled) {
-                    doCheckpoint();
-                } else {
-                    synchronized (this) {
-                        scheduledCheckpointProgress.nextCheckpointNanos(MILLISECONDS.toNanos(nextCheckpointInterval()));
-                    }
-                }
+                doCheckpoint();
             }
 
             // Final run after the cancellation.
-            if (checkpointsEnabled && !shutdownNow) {
+            if (!shutdownNow) {
```

Comment:
Is there a "disable checkpoint on shutdown" flag somewhere? It's essential for data consistency in certain cases

Reply:
Discussed in person, decided to remove **Checkpointer#skipCheckpointOnNodeStop** for now.